src/cloud-api-adaptor/podvm/files/etc/systemd/system/attestation-agent.service

```diff
@@ -1,11 +1,11 @@
 [Unit]
 Description=Attestation Agent TTRPC API Server
-After=network.target process-user-data.service
+After=network.target cloud-final.service process-user-data.service
```

Comment:
the `cloud-final.service` seems only enabled for s390x from https://github.com/confidential-containers/cloud-api-adaptor/blob/main/src/cloud-api-adaptor/podvm-mkosi/mkosi.presets/system/mkosi.finalize.chroot#L6

for amd64 build `cloud-final.service` is disable, so maybe we need another way to add this dependency?

Reply:
I think this can be done using a drop-in file? we can have a specific `mkosi.skeleton-s390x` folder?

Reply:
I think this is provider specific bot not architecture specific. We'll still need cloud-final.service for libvirt provider on x86. But don't need it for azure and aws providers because these providers are using process-user-data instead of cloud-init.

Reply:
Ok, I understand libvirt is a non-tee environment for testing only? then we could have `mkosi.skeleton-libvirt` and `mkosi.skeleton-ibmse`?

Reply:
Libvirt could be a tee environment  on s390x, this is also a supported case.

Reply:
But, yes, I think we might use mkosi.skeleton to overwrite the services file. Let me try...

Reply:
I think we have to make up our mind what kind of images we want to build. So far, we had x86_64 images [debug, production] and an exception clause for s390x:

https://github.com/confidential-containers/cloud-api-adaptor/blob/e1712af119297dd37872b795bc155de0f72b8e6c/src/cloud-api-adaptor/podvm-mkosi/Makefile#L54-L65

libvirt x86_64 doesn't seem to be supported on mkosi currently, and we should maybe address it in a later PR?

Reply:
There is no providers specific code in mkosi Makefile. 
For x86_64, there is `[debug, release]` image.
For s390x, there is `[debug, se-release] image. I just added a new type `se-debug` in PR https://github.com/confidential-containers/cloud-api-adaptor/pull/1877.
Before the remote attestation support, libvirt should worked well both on s390x and x86_64. With attestation, it breaks on mkosi image for libvirt because mkosi image is fedora based, and the process-user-data failure causes the attestation configuration breaks. Ideally, we should not add provider specific logic in mkosi, but honestly I do not think about a ideal solution yet. If we use mkosi.skeleton.s390x, will cause libvirt provider won't work on x86_64 for fedora. Still thinking about that...

Reply:
Per systemd's document, if service A defined as After= service B, it will still start if service B is disabled.
So I think systemd's `After=` constraints will not block the `kata-agent`, `attestation-agent`, `cdh` service's startup if the cloud-init service is not enabled. Maybe we can have a try on azure with the code change in this PR. @mkulke

Reply:
Indeed, that might be an option. it's not pretty, but we can add a comment that cloud-init is not present on read-only rootfs's. I'll test